### PR TITLE
Fully adopt sbt-typelevel, migrate site to Laika

### PR DESCRIPTION
**Pull request by armanbilge**
These build updates fully adopt the latest features in sbt-typelevel. This will make it easier to add support for Scala Native shortly.

I also noticed your current website is broken (code doesn't render and the page cuts off very early): https://typelevel.org/cats-parse/

So I migrated it to the sbt-typelevel-site plugin which uses Laika. This simplifies the build config. Preview here: https://armanbilge.github.io/cats-parse

Let me know if you have any questions, thanks!

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -12,27 +11,9 @@ ThisBuild / developers += tlGitHubDev("johnynek", "P. Oscar Boykin")
...
-ThisBuild / githubWorkflowBuild := Seq(
-  WorkflowStep.Run(
-    List(
-      """sbt ++${{ matrix.scala }} fmtCheck \
-        |    "++${{ matrix.scala }} test" \
-        |    "++${{ matrix.scala }} doc" \
-        |    "++${{ matrix.scala }} mimaReportBinaryIssues"""".stripMargin
-    )
-  )
-)
```
Comment (armanbilge) [inline build.sbt:25]: sbt-typelevel generates these steps automatically.

File: build.sbt [hunk 2]
```diff
@@ -12,27 +11,9 @@ ThisBuild / developers += tlGitHubDev("johnynek", "P. Oscar Boykin")
...
 
 ThisBuild / githubWorkflowAddedJobs ++= Seq(
-  WorkflowJob(
-    id = "build-docs",
-    name = "Build docs",
-    scalas = List("2.13.8"),
-    steps = List(WorkflowStep.Checkout) ++ WorkflowStep.SetupJava(
-      githubWorkflowJavaVersions.value.toList
-    ) ++ githubWorkflowGeneratedCacheSteps.value ++ List(WorkflowStep.Sbt(List("docs/mdoc")))
-  ),
```
Comment (armanbilge) [inline build.sbt:35]: Also generated automatically.

File: build.sbt [hunk 3]
```diff
@@ -43,90 +24,28 @@ ThisBuild / githubWorkflowAddedJobs ++= Seq(
...
-      githubWorkflowGeneratedCacheSteps.value :+
-      WorkflowStep.Sbt(List("docs/makeSite")) :+
-      WorkflowStep.Use(
-        UseRef.Public("JamesIves", "github-pages-deploy-action", "3.7.1"),
-        params = Map(
-          "GITHUB_TOKEN" -> "${{ secrets.GITHUB_TOKEN }}",
-          "BRANCH" -> "gh-pages",
-          "FOLDER" -> "docs/target/site"
-        )
-      )
```
Comment (armanbilge) [inline build.sbt:67]: Also generated automatically.

File: build.sbt [hunk 4]
```diff
@@ -43,90 +24,28 @@ ThisBuild / githubWorkflowAddedJobs ++= Seq(
...
-      scalacOptions ++= {
-        val isScala211 = CrossVersion.partialVersion(scalaVersion.value).contains((2, 11))
-        if (isScala211) List("-target:jvm-1.8") else List("-release", "8")
-      },
-      // Suppresses problems with Scaladoc @throws links
-      Compile / doc / scalacOptions ++= Seq("-no-link-warnings"),
-      javacOptions ++= Seq("--release", "8")
-    )
-  case _ => Def.settings()
-}
```
Comment (armanbilge) [inline build.sbt:85]: These settings were upstreamed.

File: build.sbt [hunk 5]
```diff
@@ -43,90 +24,28 @@ ThisBuild / githubWorkflowAddedJobs ++= Seq(
...
+lazy val root = tlCrossRootProject.aggregate(core, bench)
 
-lazy val root = project
-  .in(file("."))
-  .aggregate(core.jvm, core.js, bench)
-  .enablePlugins(NoPublishPlugin)
-  .settings(scalaVersion := "2.13.8")
```
Comment (armanbilge) [inline build.sbt:91]: The `tlCrossRootProject` configures the CI matrix with separate jobs for JVM and JS (and eventually Native).

File: build.sbt [hunk 6]
```diff
@@ -43,90 +24,28 @@ ThisBuild / githubWorkflowAddedJobs ++= Seq(
...
+lazy val docs =
+  project.in(file("site")).enablePlugins(TypelevelSitePlugin).dependsOn(core.jvm, bench)
```
Comment (armanbilge) [inline build.sbt:35]: The `TypelevelSitePlugin` is documented here: https://typelevel.org/sbt-typelevel/site.html

To preview locally use `docs/tlSitePreview` in sbt.

File: build.sbt [hunk 7]
```diff
@@ -135,31 +54,23 @@ lazy val core = crossProject(JSPlatform, JVMPlatform)
...
-          ProblemFilters.exclude[FinalClassProblem]("cats.parse.RadixNode"),
-          ProblemFilters.exclude[IncompatibleMethTypeProblem]("cats.parse.Parser#State.error_="),
-          ProblemFilters.exclude[IncompatibleMethTypeProblem]("cats.parse.RadixNode.this"),
-          ProblemFilters.exclude[IncompatibleResultTypeProblem]("cats.parse.Parser#State.error"),
           ProblemFilters.exclude[DirectMissingMethodProblem]("cats.parse.Parser#Error.fromProduct"),
           ProblemFilters.exclude[IncompatibleResultTypeProblem]("cats.parse.Parser#Error.unapply"),
-          ProblemFilters.exclude[MissingTypesProblem]("cats.parse.Parser$Error$")
+          ProblemFilters.exclude[MissingTypesProblem]("cats.parse.Parser$Error$"),
+          ProblemFilters.exclude[IncompatibleResultTypeProblem]("cats.parse.Parser#Error.unapply"),
+          ProblemFilters.exclude[DirectMissingMethodProblem]("cats.parse.Parser#Error.fromProduct")
```
Comment (armanbilge) [inline build.sbt:73]: MiMa had some improvements for Scala 3 so we can remove some unnecessary filters.

**Review by johnynek — APPROVED** [review]

**Comment by johnynek** [comment]
Thanks so much for doing this!

**Review by regadas — APPROVED** [review]
👍 I only realized last night that most of these things could be clean up, thx!

I like that every project under typelevel has the same site layout.

**Pull request merged**